Block/Adminhtml/Analytics/Index.php

```diff
@@ -0,0 +1,422 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Block\Adminhtml\Analytics;
+
+use Algolia\AlgoliaSearch\Helper\AnalyticsHelper;
+use Magento\Backend\Block\Template;
+use Magento\Backend\Block\Template\Context;
+use Magento\Catalog\Model\ResourceModel\Category\CollectionFactory as CategoryCollection;
+use Magento\Catalog\Model\ResourceModel\Product\CollectionFactory as ProductCollection;
+use Magento\Cms\Model\ResourceModel\Page\CollectionFactory as PageCollection;
+use Magento\Framework\Stdlib\DateTime\TimezoneInterface;
+
+class Index extends Template
+{
+    const LIMIT_RESULTS = 5;
+    const DEFAULT_TYPE = 'products';
+    const DEFAULT_RETENTION_DAYS = 7;
+
+    /** @var Context */
+    private $backendContext;
+
+    /** @var AnalyticsHelper */
+    private $analyticsHelper;
+
+    /** @var TimezoneInterface */
+    private $dateTime;
+
+    /** @var ProductCollection */
+    private $productCollection;
+
+    /** @var CategoryCollection */
+    private $categoryCollection;
+
+    /** @var PageCollection */
+    private $pageCollection;
+
+    private $analyticsParams = [];
+
+    /**
+     * Index constructor.
+     * @param Context $context
+     * @param AnalyticsHelper $analyticsHelper
+     * @param TimezoneInterface $dateTime
+     * @param ProductCollection $productCollection
+     * @param CategoryCollection $categoryCollection
+     * @param PageCollection $pageCollection
+     * @param array $data
+     */
+    public function __construct(
+        Context $context,
+        AnalyticsHelper $analyticsHelper,
+        TimezoneInterface $dateTime,
+        ProductCollection $productCollection,
+        CategoryCollection $categoryCollection,
+        PageCollection $pageCollection,
```

Comment:
`ProductCollection`, `CategoryCollection`, `PageCollection` are injected to enrich / hydrate popular results in `getPopularResults()` method. How about extracting this logic to a standalone class (like `PopularResultsDataHydrator` :)) where the logic can sit and you could have a single method for each data type (product, category, page)? WDYT?